third_party/airflow/armada/operators/utils.py

```diff
@@ -11,10 +11,32 @@
 
 
 class JobState(Enum):
-    SUCCEEDED = 1
-    FAILED = 2
-    CANCELLED = 3
-    JOB_ID_NOT_FOUND = 4
+    SUBMITTED = 0
+    DUPLICATE_FOUND = 1
+    RUNNING = 2
+    FAILED = 3
+    SUCCEEDED = 4
+    CANCELLED = 5
+    JOB_ID_NOT_FOUND = 6
+    CONNECTION_ERR = 7
+
+
+_pb_to_job_state = {
+    jobservice_pb2.JobServiceResponse.SUBMITTED: JobState.SUBMITTED,
+    jobservice_pb2.JobServiceResponse.DUPLICATE_FOUND: JobState.DUPLICATE_FOUND,
+    jobservice_pb2.JobServiceResponse.RUNNING: JobState.RUNNING,
+    jobservice_pb2.JobServiceResponse.FAILED: JobState.FAILED,
+    jobservice_pb2.JobServiceResponse.SUCCEEDED: JobState.SUCCEEDED,
+    jobservice_pb2.JobServiceResponse.CANCELLED: JobState.CANCELLED,
+    jobservice_pb2.JobServiceResponse.JOB_ID_NOT_FOUND: JobState.JOB_ID_NOT_FOUND,
+    # NOTE(Clif): For whatever reason CONNECTION_ERR is not present in the
+    # generated protobuf.
+    7: JobState.CONNECTION_ERR,
+}
+
+
+def job_state_from_pb(state) -> JobState:
```

Comment:
```suggestion
def job_state_from_pb(state: str) -> JobState:
```

Not sure if this is correct.

Reply:
But we aim to have typing in the python code.